components/table/README.md

```diff
@@ -299,3 +299,87 @@ If your table supports row selection, apply the `selected` attribute to `<tr>` r
   <td>this row is selected</td>
 </tr>
 ```
+
+## Table Header [d2l-table-header]
+
+The `d2l-table-header` component can be placed in the `d2l-table-wrapper`'s `header` slot to provide a selection summary, a slot for `d2l-selection-action`s, and overflow-group behaviour.
+
+<!-- docs: demo live name:d2l-table-header autoSize:false size:medium -->
+```html
+<script type="module">
+  import '@brightspace-ui/core/components/selection/selection-action.js';
+  import '@brightspace-ui/core/components/selection/selection-input.js';
+  import '@brightspace-ui/core/components/selection/selection-select-all.js';
+  import '@brightspace-ui/core/components/table/table-header.js';
+  import { html, LitElement } from 'lit';
+  import { tableStyles } from '@brightspace-ui/core/components/table/table-wrapper.js';
+
+  class MyTableWithHeaderElem extends LitElement {
+
+    static get properties() {
+      return {
+        _checked: { state: true }
+      }
+    }
+
+    static get styles() {
+      return tableStyles;
+    }
+
+    constructor() {
+      super();
+      this._checked = [ true, false ];
```

Comment:
This is fairly basic "checked" logic for the readme - I could instead model the more standard pattern of something like `this._data = [ {selected: true, ...}, ... ]`, but I thought that was maybe overkill here.